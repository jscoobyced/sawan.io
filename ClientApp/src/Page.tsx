import React from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
import { BlogEditHoc } from './components/blog/BlogEditHoc';
import { BlogHoc } from './components/blog/BlogHoc';
import { CandleChartPageHoc } from './components/charts/CandleChartPageHoc';
import { About } from './components/main/About';
import { HomeHoc } from './components/main/HomeHoc';
import { Layout } from './components/main/Layout';
import { Resume } from './components/main/Resume';
import { ContentServiceFactory } from './services/ContentServiceFactory';
import { IContentService } from './services/IContentService';
import { Language, MainContent } from './services/Models';
import { HtmlUtils } from './utils/HtmlUtils';

export interface PageState {
    allContent: MainContent;
    isSignedIn: boolean;
}

export class Page extends React.Component<{}, PageState> {

    private readonly contentService: IContentService;

    public constructor(props: {}) {
        super(props);
        this.contentService = ContentServiceFactory.GetContentService();
        this.state = {
            allContent: this.contentService.getDefaultMainContent(),
            isSignedIn: false
        };
    }

    public componentDidMount() {
        this.contentService.getMainContent(Language.English)
            .then(allContent => {
                this.setState({ allContent });
            });
    }

    public render() {
        if (HtmlUtils.queryString().indexOf('full') > 0) {
            // hard-coded resume. Will need something more dynamic
            return <Resume />;
        }
        return <BrowserRouter children={this.routes()} basename={HtmlUtils.baseUrl()} />;
    }

    private routes() {
        return <Layout
            allContent={this.state.allContent}
            signIn={this.signIn}
            isSignedIn={this.state.isSignedIn} >
            <Route exact path='/' component={() => <HomeHoc isSignedIn={this.state.isSignedIn} />} />
            <Route path='/candle' component={CandleChartPageHoc} />
            <Route path='/health' component={About} />
            <Route path='/blog/view/:id/' component={BlogHoc} />
            <Route path='/blog/edit/:id/' component={BlogEditHoc} />
            <Route path='/about' component={About} />
            <Route path='/resume' component={Resume} />
        </Layout>;
    }

    private readonly signIn = (isSignedIn: boolean) => {
        this.setState({
            isSignedIn
        });
    }

}
