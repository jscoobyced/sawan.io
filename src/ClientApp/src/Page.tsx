import React from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
import { CandleChartPageHoc } from './components/charts/CandleChartPageHoc';
import { Home } from './components/main/Home';
import { Layout } from './components/main/Layout';
import { ContentServiceFactory } from './services/ContentServiceFactory';
import { IContentService } from './services/IContentService';
import { BlogPage, Language, MainContent } from './services/Models';
import { HtmlUtils } from './utils/HtmlUtils';

export interface PageState {
    allContent: MainContent;
    blogPage: BlogPage;
}

export class Page extends React.Component<{}, PageState> {

    private readonly contentService: IContentService;

    public constructor(props: any) {
        super(props);
        const mode = process.env.mode as string;
        this.contentService = ContentServiceFactory.GetContentService(mode);
        this.state = {
            allContent: this.contentService.getDefaultMainContent(),
            blogPage: this.contentService.getDefaultBlogPage()
        };
    }

    public componentDidMount() {
        this.contentService.getMainContent(Language.English)
            .then(allContent => {
                this.contentService.getBlogPage(0, 4)
                    .then(blogPage => {
                        this.setState({ allContent, blogPage });
                    });
            });
    }

    public render() {
        const routes = <Layout allContent={this.state.allContent} >
            <Route exact path='/' component={() => <Home blogPage={this.state.blogPage} />} />
            <Route path='/candle' component={CandleChartPageHoc} />
            <Route path='/health' component={CandleChartPageHoc} />
        </Layout>;
        return <BrowserRouter children={routes} basename={HtmlUtils.baseUrl()} />;
    }
}
