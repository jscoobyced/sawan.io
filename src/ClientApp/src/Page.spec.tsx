import { shallow } from 'enzyme';
import React from 'react';
import { Page } from './Page';

test('Page content is preserved', () => {
    const page = shallow(<Page />);
    expect(page).toMatchSnapshot();
});
