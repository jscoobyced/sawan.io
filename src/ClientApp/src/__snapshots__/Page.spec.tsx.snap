// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page content is preserved 1`] = `
<BrowserRouter
  basename=""
>
  <Layout
    allContent={
      Object {
        "homeContent": Object {
          "article": Array [
            "Grand opening of sawan.io.<br />This website is about my personal training on various web-technologies: TypeScript, webpack, ReactJS, Secure HTTP headers... It includes a simple (and a bit naive :) ) crypto-currency analyser: it can suggest you to sell or buy. Use at your own risk, it is amateur predictions, you have been warned :)<br />There will be also an application to track health metrics (weight, % body fat, vascular fat ratio... Or not, this will be my next project :)<br /> This project is open-sourced on <a target='_blank' href='https://github.com/jscoobyced/sawan.io'>Github</a>. Enjoy and have fun.",
            "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.<br />Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Commodo quis imperdiet massa tincidunt. Non enim praesent elementum facilisis leo. Adipiscing diam donec adipiscing tristique. Velit euismod in pellentesque massa placerat. Interdum velit euismod in pellentesque. Phasellus vestibulum lorem sed risus ultricies tristique nulla. Egestas erat imperdiet sed euismod nisi porta lorem mollis aliquam. Nulla facilisi etiam dignissim diam quis enim lobortis scelerisque fermentum. Morbi tristique senectus et netus et malesuada fames.<br />Nisi vitae suscipit tellus mauris a diam maecenas. Sed risus ultricies tristique nulla aliquet. Sagittis eu volutpat odio facilisis mauris sit amet. Tempor orci eu lobortis elementum nibh tellus molestie nunc non. Turpis nunc eget lorem dolor.",
            "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Commodo quis imperdiet massa tincidunt. Non enim praesent elementum facilisis leo. Adipiscing diam donec adipiscing tristique. Velit euismod in pellentesque massa placerat. Interdum velit euismod in pellentesque. Phasellus vestibulum lorem sed risus ultricies tristique nulla. Egestas erat imperdiet sed euismod nisi porta lorem mollis aliquam. Nulla facilisi etiam dignissim diam quis enim lobortis scelerisque fermentum. Morbi tristique senectus et netus et malesuada fames.<br />Nisi vitae suscipit tellus mauris a diam maecenas. Sed risus ultricies tristique nulla aliquet. Sagittis eu volutpat odio facilisis mauris sit amet. Tempor orci eu lobortis elementum nibh tellus molestie nunc non. Turpis nunc eget lorem dolor. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Commodo quis imperdiet massa tincidunt. Non enim praesent elementum facilisis leo. Adipiscing diam donec adipiscing tristique. Velit euismod in pellentesque massa placerat. Interdum velit euismod in pellentesque. Phasellus vestibulum lorem sed risus ultricies tristique nulla.<br />Egestas erat imperdiet sed euismod nisi porta lorem mollis aliquam. Nulla facilisi etiam dignissim diam quis enim lobortis scelerisque fermentum. Morbi tristique senectus et netus et malesuada fames. Nisi vitae suscipit tellus mauris a diam maecenas. Sed risus ultricies tristique nulla aliquet.",
          ],
          "articleTitle": Array [
            "Latest news - 2018-11-15",
            "Lorem Ipsum",
            "Some other news",
          ],
        },
        "navigationMenuContent": Object {
          "about": "About",
          "applications": "Applications",
          "cryptoCurrency": "Cryptocurrency",
          "healthMonitor": "Health Monitor",
          "home": "Home",
          "websiteName": "sawan.io",
        },
      }
    }
  >
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/candle"
    />
    <Route
      component={[Function]}
      path="/health"
    />
  </Layout>
</BrowserRouter>
`;
